Diary/Resource/Common/DiaryItemManager.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  DiaryItemManager.swift
+//  Diary
+//
+//  Created by SummerCat and som on 2022/12/28.
+//
+
+import Foundation
+
+final class DiaryItemManager {
+    static let shared = DiaryItemManager()
+    private var diaryItem: DiaryModel?
+    
+    func saveDiaryWith(title: String, body: String) {
+        if CoreDataManager.shared.fetchDiary(with: diaryItem?.id) == nil {
+            generateDiary()
+            updateDiaryTo(title: title, body: body)
+            
+            if !validate(diaryItem: diaryItem) { return }
+            
+            CoreDataManager.shared.insertDiary(diaryItem)
+            let id = CoreDataManager.shared.fetchID(of: diaryItem)
+            diaryItem?.id = id
+        } else {
+            updateDiaryTo(title: title, body: body)
+            CoreDataManager.shared.updateDiary(diaryItem)
+        }
+    }
+    
+    private func generateDiary() {
+        diaryItem = DiaryModel(title: Namespace.empty,
+                               body: Namespace.empty,
+                               createdAt: Date())
+    }
+    
+    private func updateDiaryTo(title: String, body: String) {
+        diaryItem?.title = title
+        diaryItem?.body = body
+    }
+    
+    private func validate(diaryItem: DiaryModel?) -> Bool {
+        if diaryItem?.title == Placeholder.title, diaryItem?.body == Namespace.empty { return false }
+        else if diaryItem?.title == Namespace.empty, diaryItem?.body == Placeholder.body { return false }
+        else if diaryItem?.title == Placeholder.title, diaryItem?.body == Placeholder.body { return false }
+        else if diaryItem?.title == Namespace.empty, diaryItem?.body == Namespace.empty { return false }
+        else { return true }
+    }
```

Comment:
음.. 이건 개인의 취향 차이 일수도 있는데, 저는 잘 안읽히네요.
아래와 같이 바꾸면 어떨까요?
```suggestion
    private func isValid(_ diaryItem: DiaryModel?) -> Bool {
        return isTitleValid(diaryItem?.title) && isBodyValid(diaryItem?.body)
    }
    
    private func isTitleValid(_ title: String?) -> Bool {
        return title != Placeholder.title && title != Namespace.empty
    }
    
    private func isBodyValid(_ body: String?) -> Bool {
        return body != Placeholder.body && body != Namespace.empty
    }
```